tensordict/utils.py

```diff
@@ -1769,27 +1769,29 @@ def _check_keys(
     )
     # TODO: compile doesn't like set() over an arbitrary object
     if is_compiling():
-        keys = {k for k in keys}  # noqa: C416
+        keys_set = {k for k in keys}  # noqa: C416
     else:
-        keys: set[str] = set(keys)
+        keys_set: set[str] = set(keys)
```

Comment:
Out of curiosity, is it much more efficient using `set` rather than using always the other option? Or there is another reason?

Reply:
torch.compile used to not understand set() that's all. I should check if it's still the case